pallets/oracle/src/lib.rs

```diff
@@ -72,6 +74,13 @@ pub mod pallet {
 			OracleKey,
 			TimestampedValue<Self::UnsignedFixedPoint, Self::Moment>,
 		>;
+
+		// #[cfg(feature = "testing-utils")]
+		type DataFeedProvider: orml_oracle::DataFeeder<
+			OracleKey,
+			TimestampedValue<Self::UnsignedFixedPoint, Self::Moment>,
+			Self::AccountId
+		>;
 	}
```

Comment:
Instead of adding a separate parameter I would have expected we just conditionally define the DataProvider. Something like
```
// Outside of the definition of the Config trait
cfg_if::cfg_if! {
	if #[cfg(feature = "testing-utils")] {
		type DataProvider = DataProviderExtended<
			OracleKey,
			TimestampedValue<Self::UnsignedFixedPoint, Self::Moment>,
		> + orml_oracle::DataFeeder<
			OracleKey,
			TimestampedValue<Self::UnsignedFixedPoint, Self::Moment>,
			Self::AccountId
		>;
	} else {
       type DataProvider: DataProviderExtended<
			OracleKey,
			TimestampedValue<Self::UnsignedFixedPoint, Self::Moment>,
		>;
	}
}
```
and then inside the Config trait we use the previously defined DataProvider.
The code probably does not work like that but just saying that I would like us to try to use some macro magic for changing that type conditionally. Do you think this could work @cr4pt0?

Reply:
@ebma 
I would like to separate responsibility for `DataProviderExtended` and` orml_oracle::DataFeeder` to different type pallet configuration. 

This way will be more easy to mock 
```
#[cfg(feature = "testing-utils")]
		type DataFeedProvider: orml_oracle::DataFeeder<
			OracleKey,
			TimestampedValue<Self::UnsignedFixedPoint, Self::Moment>,
			Self::AccountId,
		>;
```
For example:
DIA pallet do not implement `orml_oracle::DataFeeder` trait.
In case we separate type configuration we can easy create other struct(Adapter) to extend DIA pallet ONLY with `orml_oracle::DataFeeder` functionality .

If we combine 2 configuration (`orml_oracle::DataFeeder` and `DataProviderExtended`) in one type
there is no way to implement directly on DIA pallet `orml_oracle::DataFeeder`

So in that case require more efforts to implement new wrapper(Adapter) o implement both (`orml_oracle::DataFeeder` and reimplement `DataProviderExtended`)

So i just want to make it more flexible and easy configure for any set of pallets.